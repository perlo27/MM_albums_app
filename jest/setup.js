import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { FluidNavigator } from 'react-navigation-fluid-transitions';

Enzyme.configure({ adapter: new Adapter() });

global.fetch = require('jest-fetch-mock');

jest.mock('react-navigation-fluid-transitions', () => ({
  'Transition': 'Transition',
  'FluidNavigator': jest.fn()
}));

jest.mock('native-base', () => ({
  'Header': 'Header',
  'Body': 'Body',
  'Left': 'Left',
  'Button': 'Button',
  'Icon': 'Icon',
}));