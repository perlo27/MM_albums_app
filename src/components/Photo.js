import React, { Component } from 'react';
import { View, Image } from 'react-native';
import { connect } from 'react-redux';
import { photo } from './propTypes';


@connect(({ entities: { photos } }, { navigation: { state: { params } } }) => {
  console.log('params', params);
  return {
    photo: photos[params.albumId].find(({ id }) => id === params.id),
  };
})
export class Photo extends Component {

  static propTypes = { photo };

  render() {
    console.log(this.props);
    return <Image style={{ width: '100%', height: '50%' }} source={{ uri: this.props.photo.url }} />;
  }
}
