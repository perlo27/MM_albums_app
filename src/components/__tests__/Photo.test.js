import React from 'react';
import renderer from 'react-test-renderer';
import { Photo } from '../Photo';

import { store, navigation, storeWithoutPhoto } from '../../../jest/moks';

jest.mock('../Header', () => ({
  'Header': 'Header'
}));



describe('Photo', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(<Photo navigation={navigation} store={store} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render without image', () => {
    const tree = renderer
      .create(<Photo navigation={navigation} store={storeWithoutPhoto} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

});