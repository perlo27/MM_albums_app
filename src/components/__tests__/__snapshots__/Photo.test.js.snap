// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Photo should render correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Transition
    anchor="image1"
  >
    <Header
      navigation={
        Object {
          "goBack": [MockFunction],
          "navigate": [MockFunction],
          "state": Object {
            "params": Object {
              "albumId": 1,
              "id": 1,
            },
          },
        }
      }
      title="Test"
    />
  </Transition>
  <Transition
    shared="image1"
  >
    <Image
      source={
        Object {
          "uri": "url",
        }
      }
      style={
        Object {
          "height": 1264,
          "width": 750,
        }
      }
    />
  </Transition>
</View>
`;

exports[`Photo should render without image 1`] = `null`;
